code/modules/assembly/holder.dm

```diff
@@ -13,200 +13,201 @@
 	var/obj/item/assembly/a_left = null
 	var/obj/item/assembly/a_right = null
 
-	proc/attach(var/obj/item/D, var/obj/item/D2, var/mob/user)
-		return
-
-	proc/process_activation(var/obj/item/D)
-		return
-
-
-
-	IsAssemblyHolder()
-		return 1
-
-	Destroy()
-		if(a_left)
-			a_left.holder = null
-		if(a_right)
-			a_right.holder = null
-		return ..()
-
-	attach(var/obj/item/D, var/obj/item/D2, var/mob/user)
-		if((!D)||(!D2))	return 0
-		if((!isassembly(D))||(!isassembly(D2)))	return 0
-		if((D:secured)||(D2:secured))	return 0
-		if(!D.remove_item_from_storage(src))
-			if(user)
-				user.remove_from_mob(D)
-			D.loc = src
-		if(!D2.remove_item_from_storage(src))
-			if(user)
-				user.remove_from_mob(D2)
-			D2.loc = src
-		D:holder = src
-		D2:holder = src
-		a_left = D
-		a_right = D2
-		name = "[D.name]-[D2.name] assembly"
-		update_icon()
-		return 1
-
-
+/obj/item/assembly_holder/proc/attach(var/obj/item/D, var/obj/item/D2, var/mob/user)
+	return
+
+/obj/item/assembly_holder/proc/process_activation(var/obj/item/D)
+	return
+
+/obj/item/assembly_holder/IsAssemblyHolder()
+	return TRUE
+
+/obj/item/assembly_holder/Destroy()
+	if(a_left)
+		a_left.holder = null
+	if(a_right)
+		a_right.holder = null
+	return ..()
+
+/obj/item/assembly_holder/attach(var/obj/item/D, var/obj/item/D2, var/mob/user)
+	if(!D || !D2)
+		return FALSE
+	if(!isassembly(D) || !isassembly(D2))
+		return FALSE
+	if(D:secured || D2:secured)
+		return FALSE
+	if(!D.remove_item_from_storage(src))
+		if(user)
+			user.remove_from_mob(D)
+		D.loc = src
+	if(!D2.remove_item_from_storage(src))
+		if(user)
+			user.remove_from_mob(D2)
+		D2.loc = src
+	D:holder = src

```

Comment:
Okay so I have no idea what this `holder` variable is referring to. Anyone know?